includes/abilities-api/class-wp-ability.php

```diff
@@ -314,6 +314,10 @@ public function has_permission( array $input = array() ) {
 			return true;
 		}
 
+		if ( empty( $this->get_input_schema() ) && null === $input ) {
+			return call_user_func( $this->permission_callback );
+		}
```

Comment:
Related to my previous comment, we should not allow any unvalidated input to be sent through. In other words, this clause should only be dependent on whether there's an input schema.
```suggestion
		if ( empty( $this->get_input_schema() ) ) {
			return call_user_func( $this->permission_callback );
		}
```